src/apps/service/apps/de/listings.clj

```diff
@@ -134,13 +140,32 @@
   [_ workspace params]
   (list-shared-apps workspace (workspace-favorites-app-category-index) params))
 
+(defn format-featured-apps-category
+  "Formats the virtual group for apps that are certified as blessed"
+  [user _ params]
+   {:system_id de-system-id
+    :id        featured-apps-id
+    :name      "Featured Apps"
+    :is_public false
+    :total     (future (count-apps-for-user nil nil {:app-ids (app-ids->certified-ids-set (:username user) (:app-ids params))}))})
+
+(defn list-featured-apps
+  [user workspace params]
+   (let [blessed-ids (app-ids->certified-ids-set (:username user) (:app-ids params))]
```

Comment:
Ideally we wouldn't have to hit the metadata service twice per listing to get the filtered list of blessed IDs.
Unfortunately I don't see another precedent for doing that with these virtual groups.

Maybe we could add an `augment-virtual-group-listing-params` function which could augment the listing params once more to update the `:app-ids` param with the filtered IDs, but for now it would only do that if the group ID is this Featured Apps category. It could be called in `list-apps-in-virtual-group`, just before the listing and format functions are called. Then both the listing and format functions could just use `(:app-ids params)` 🤔 

Actually, we should probably pass the entire `params` map to the count function? It could make the total count more accurate, since it could be affected by other listing params that might filter the listing even further. For example, if the user wants to view only VICE apps in the Featured category.

Reply:
Ah, I think you're correct about the count.  Thanks for catching that!

I actually started to do what you're suggesting but added a `:listing-ids` param to params so anything referencing `:app-ids` wouldn't be affected, but ran into issues so I reverted that.  If I try augmenting the params before any of the listing or format functions are called though like you suggested, it might work. I'll give that a shot!